src/nvim/popupmnu.c

```diff
@@ -136,7 +136,10 @@ void pum_display(pumitem_T *array, int size, int selected, bool array_changed,
       } else {
         ui_call_popupmenu_select(selected);
       }
-      return;
+      pum_height = ui_pum_height();
```

Comment:
Only gathering the height here is too restrictive. We should allow the UI to set the height after receiving the items, which might change the height it needs.

Reply:
I e `pum_get_height()` should always use the most recent value set by the UI.

Reply:
@bfredl For now, I changed `pum_get_height()` to call `ui_pum_get_height()`. I think that there is a problem if `<PageUp>` etc. call `pum_get_height()` before the UI calls `nvim_ui_pum_set_height()`. How do I synchronize? Can you give me some implementation hint? Should I call `LOOP_PROCESS_EVENTS_UNTIL(& main_loop, NULL, -1, is_pum_set_height_done())` with `pum_get_height()`? A deadlock occurred when trying to synchronize using the above method immediately after `ui_call_popupmenu_show()`.

Reply:
There should be no need to synchronize anything. As long as the UI sends the `nvim_ui_pum_set_height()`  before `nvim_input('<PageUp>')` it should be processed before. If an indirect mapping or something uses a slightly outdated value, it is not a catastrophe. Let's not make it more complicated than needed.

Reply:
Though you could mark nvim_ui_pum_set_height() as `NVIM_API_FAST`, so the events cannot get reordered internally, in case nvim was CPU-blocked by an unrelated timer, or something.